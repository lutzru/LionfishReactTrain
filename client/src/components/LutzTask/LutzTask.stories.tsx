import { action } from '@storybook/addon-actions'
import { text, withKnobs } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import 'utils/ i18n'
import LutzTask, { LutzTaskProps } from './LutzTask'

const stories = storiesOf('LutzTask', module)

const props: LutzTaskProps = {
    onTaskDateChanged: action('onTaskDateChanged'),
    taskDate: '11.12.2020',
}

stories.addDecorator(withKnobs)
stories.add('default view', () => <LutzTask {...props} />)

stories.addDecorator(withKnobs)
stories.add('default view (interactive)', () => (
    <LutzTask onTaskDateChanged={action('onTaskDateChanged')} taskDate={text('taskDate', '')} />
))
