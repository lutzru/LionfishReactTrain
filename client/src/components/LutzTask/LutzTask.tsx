import { Box, Grid, TextField } from '@material-ui/core'
import * as React from 'react'

export interface LutzTaskProps {
    onTaskDateChanged: (searchText: string) => void
    taskDate: string
}

const LutzTask = (props: LutzTaskProps): JSX.Element => {
    return (
        <Box padding={3}>
            <Grid container alignItems="center" spacing={4}>
                <Grid item>
                    <TextField
                        id="taskDate"
                        label="Datum:"
                        value={props.taskDate}
                        onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
                            props.onTaskDateChanged(e.target.value)
                        }}
                    />
                </Grid>
                <Grid item>Prio:</Grid>
                <Grid item>Text:</Grid>
                <Grid item>Status:</Grid>
            </Grid>
        </Box>
    )
}

export default LutzTask
