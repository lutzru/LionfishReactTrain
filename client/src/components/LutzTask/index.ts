import LutzTask from './LutzTask'

export default LutzTask
