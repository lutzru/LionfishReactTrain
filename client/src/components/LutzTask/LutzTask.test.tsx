import { TextField } from '@material-ui/core'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import * as React from 'react'
import LutzTask, { LutzTaskProps } from './LutzTask'

jest.mock('react-i18next', () => ({
    useTranslation: (): unknown => ({ t: (key: string): string => key }),
}))
jest.mock('react', () => ({
    // @ts-ignore
    ...jest.requireActual('react'),
    useState: jest.fn(),
}))

const props: LutzTaskProps = {
    onTaskDateChanged: jest.fn(),
    taskDate: '01.02.1987',
}

describe('LutzTask Component', () => {
    it('should render correctly with enzyme', () => {
        const component = shallow(<LutzTask {...props} />)

        expect(toJson(component)).toMatchSnapshot()
    })
    it('should handle a click correctly', () => {
        const onClickMock = jest.fn()
        const component = shallow(<LutzTask onTaskDateChanged={onClickMock} taskDate={'01.02.1987'} />)

        component.find(TextField).simulate('change', { target: { value: 'Test' } })
        expect(onClickMock).toHaveBeenCalledTimes(1)
    })
})
