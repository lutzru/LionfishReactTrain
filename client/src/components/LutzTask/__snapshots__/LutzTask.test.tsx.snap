// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LutzTask Component should render correctly with enzyme 1`] = `
<Styled(MuiBox)
  padding={3}
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
    spacing={4}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(TextField))
        id="taskDate"
        label="Datum:"
        onChange={[Function]}
        value="01.02.1987"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      Prio:
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      Text:
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      Status:
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Styled(MuiBox)>
`;
